app/src/main/kotlin/at/bitfire/davdroid/di/CoroutineDispatchersModule.kt

```diff
@@ -40,4 +50,24 @@ class CoroutineDispatchersModule {
     @MainDispatcher
     fun mainDispatcher(): CoroutineDispatcher = Dispatchers.Main
 
+    /**
+     * A dispatcher for background sync operations. They're not run on [ioDispatcher] because there can
+     * be many long-blocking operations at the same time which shouldn't never block other I/O operations
+     * like database access for the UI.
+     */
+    @Provides
+    @Singleton
+    @SyncDispatcher
+    fun syncDispatcher(): CoroutineDispatcher =
+        ThreadPoolExecutor(
+            /* corePoolSize = */ 0,
+            /* maximumPoolSize = */ Runtime.getRuntime().availableProcessors(),
+            /* keepAliveTime = */ 10,
+            /* unit = */ TimeUnit.SECONDS,
+            /* workQueue = */ LinkedBlockingQueue(),
```

Comment:
Maybe we should add a bound to `LinkedBlockingQueue` ? 

From https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ThreadPoolExecutor.html

>  If there are more than corePoolSize but less than maximumPoolSize threads running, a new thread will be created only if the queue is full.

Maybe I misunderstand something, but with corePoolSize=0 and maximumPoolSize > 0 and unbounded `LinkedBlockingQueue`  the queue will never be full and so new threads will never be created?

So without a bound for `LinkedBlockingQueue` setting maximumPoolSize to anything bigger than 0 is unnecessary?

Reply:
Good point! I watched the threads with `adb shell ps -Aw` during a sync, and only 1 "SyncDispatcher" thread was created. I thought at was because there was no more concurrent work but now I think it's because the queue never becomes full.

Reply:
Next try: `Dispatchers.IO.limitedParallelism(Runtime.getRuntime().availableProcessors())`

According to [Elasticity for limited parallelism](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-dispatchers/-i-o.html), it does exactly what we want and it does _not_ influence the default Dispatchers.IO threads.